lib/Backend/GlobOpt.cpp

```diff
@@ -3825,32 +3826,156 @@ GlobOpt::TestAnyArgumentsSym()
     return blockData.argObjSyms->TestEmpty();
 }
 
+/*
+*   This is for scope object removal along with Heap Arguments optimization.
+*   We track several instructions to facilitate the removal of scope object.
+*   - LdSlotArr - This instr is tracked to keep track of the formals array (the dest)
+*   - InlineeStart - To keep track of the stack syms for the formals of the inlinee.
+*/
+void
+GlobOpt::TrackInstrsForScopeObjectRemoval(IR::Instr * instr)
+{
+    IR::Opnd* dst = instr->GetDst();
+    IR::Opnd* src1 = instr->GetSrc1();
+
+    if (instr->m_opcode == Js::OpCode::Ld_A && src1->IsRegOpnd())
+    {
+        AssertMsg(!src1->IsScopeObjOpnd(instr->m_func), "There can be no aliasing for scope object.");
+    }
+
+    // The following is to track formals array for Stack Arguments optimization with Formals
+    if (instr->IsStackArgsEnabled() && !this->IsLoopPrePass())
+    {
+        if (instr->m_opcode == Js::OpCode::LdSlotArr)
+        {
+            if (instr->GetSrc1()->IsScopeObjOpnd(instr->m_func))
+            {
+                AssertMsg(!instr->m_func->GetJnFunction()->GetHasImplicitArgIns(), "No mapping is required in this case. So it should already be generating ArgIns.");
+                instr->m_func->TrackFormalsArraySym(dst->GetStackSym()->m_id);
+            }
+        }
+        else if (instr->m_opcode == Js::OpCode::InlineeStart)
+        {
+            Assert(instr->m_func->IsInlined());
+            Js::ArgSlot actualsCount = instr->m_func->actualCount - 1;
+            Js::ArgSlot formalsCount = instr->m_func->GetJnFunction()->GetInParamsCount() - 1;
+
+            Func * func = instr->m_func;
+            Func * topFunc = this->func;
```

Comment:
Yes. My bad :( Fixed.

---
In reply to: [67909136](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r67909136) [](ancestors = 67909136)